openwebstart/src/main/java/com/openwebstart/proxy/windows/WindowsProxyUtils.java

```diff
@@ -77,14 +77,28 @@ private static ProxyConfigurationImpl getProxyConfiguration(final String proxySe
             proxyConfiguration.setUseHttpForHttpsAndFtp(true);
             proxyConfiguration.setUseHttpForSocks(true);
 
-            final String[] split = hosts.get(0).split(Pattern.quote(":"), 2);
-            if (split.length == 1) {
-                //TODO: Default port??
-                throw new IllegalStateException("No port defined!");
-            } else {
-                //TODO: We need to check port behavior on win
-                proxyConfiguration.setHttpHost(split[0]);
-                proxyConfiguration.setHttpPort(Integer.parseInt(split[1]));
+            String proxyString = hosts.get(0);
+            try {
+                if (!proxyString.contains("://")) {
+                    proxyString = "http://" + proxyString;
+                }
+
+                URL url = new URL(proxyString);
+                String protocol = url.getProtocol();
+                String host = url.getHost();
+                int port = url.getPort();
+
+                if (port == -1) {
+                    LOG.debug("Port missing in Proxy " + hosts.get(0));
```

Comment:
please use `proxyString` instead of `host.get(0)` in the log. This improves readability